src/main/resources/include/hail/hail.h

```diff
@@ -0,0 +1,23 @@
+#ifndef HAIL_HAIL_H
+#define HAIL_HAIL_H 1
+// hail.h - top-level header file for use by dynamically-generated C++ code
+#include "hail/NativeObj.h"
+#include "hail/NativeStatus.h"
+
+// A dynamic-generated source file will be passed down as a Java String,
+// together with some compiler options.  The combination of the compiler
+// options and the string contents will be used to generate an 80-bit
+// hash key, which will then be used both to generate the filename for the
+// .cpp and .so (or .dylib) files.
+//
+// This hashcode will also be provided as part of -DHAIL_MODULE when
+// compiling the code.
```

Comment:
I'm not sure I understand the context for these comments.  Do they still belong here?